sympy/geometry/ellipse.py

```diff
@@ -1362,6 +1357,49 @@ def __new__(cls, *args, **kwargs):
 
         raise GeometryError("Circle.__new__ received unknown arguments")
 
+    @classmethod
+    def object_from_equation(self, equation):
+        """Returns a Circle object from an equation of circle given as input.
+        If the equation of the circle is ax**2 + by**2 + gx + hy + c = 0, then
+        the input has to be of the following format: ax**2 + by**2 + gx + hy + c
+
+        Examples
+        ========
+
+        >>> from sympy import Circle
+        >>> from sympy.abc import x, y
+        >>> Circle.object_from_equation(x**2 + y**2 + 3*x + 4*y - 8)
+        Circle(Point2D(-3/2, -2), sqrt(57)/2)
+
+        """
+
+        x = var('x')
+        y = var('y')
+        if x in list(equation.free_symbols) or y in list(equation.free_symbols):
+            poly_x = poly(equation, x)
+            poly_y = poly(equation, y)
+            deg_x = degree(poly_x)
+            deg_y = degree(poly_y)
+            lead_x_coeff = LC(poly_x)
+            lead_y_coeff = LC(poly_y)
+
+            if lead_x_coeff == lead_y_coeff and (deg_x == 2 and deg_y == 2):
+                center_x = solve(diff(equation, x))[0]
+                center_y = solve(diff(equation, y))[0]
+                center = Point(center_x, center_y)
+                r_square = expand(lead_x_coeff * (x - center_x) ** 2) + expand(
+                    lead_y_coeff * (y - center_y) ** 2) - equation
+                radius = sqrt(r_square / lead_x_coeff)
```

Comment:
This could go inside the if condition, to prevent its computation in the negative case.

Reply:
@sidhantnagpal Sure, I'll do that.